fluence-faas/src/misc/config.rs

```diff
@@ -48,88 +53,109 @@ An example of the config:
+    pub fn new(name: String) -> Self {
+        Self {
+            name,
+            mem_pages_count: None,
+            logger_enabled: None,
+            imports: None,
+            wasi: None,
+        }
+    }
+}
+
+#[derive(Deserialize, Serialize, Debug, Clone, Default)]
+pub struct RawWASIConfig {
     pub envs: Option<Vec<String>>,
     pub preopened_files: Option<Vec<String>>,
     pub mapped_dirs: Option<toml::value::Table>,
 }
 
-#[derive(Deserialize, Debug)]
-pub(crate) struct RawRPCModuleConfig {
+#[derive(Deserialize, Serialize, Debug, Clone, Default)]
+pub struct RawRPCModuleConfig {
     pub mem_pages_count: Option<u32>,
     pub logger_enabled: Option<bool>,
     pub wasi: Option<RawWASIConfig>,
 }
 
-#[derive(Debug)]
-pub(crate) struct NodeConfig {
+#[derive(Debug, Clone, Default)]
+pub struct CoreModulesConfig {
+    pub core_modules_dir: Option<String>,
     pub modules_config: HashMap<String, ModuleConfig>,
     pub rpc_module_config: Option<ModuleConfig>,
 }
 
-#[derive(Debug)]
-pub(crate) struct ModuleConfig {
+#[derive(Debug, Clone, Default)]
+pub struct ModuleConfig {
     pub mem_pages_count: Option<u32>,
     pub logger_enabled: Option<bool>,
     pub imports: Option<Vec<(String, String)>>,
     pub wasi: Option<WASIConfig>,
 }
 
-#[derive(Debug)]
-pub(crate) struct WASIConfig {
+#[derive(Debug, Clone, Default)]
+pub struct WASIConfig {
     pub envs: Option<Vec<Vec<u8>>>,
     pub preopened_files: Option<Vec<String>>,
     pub mapped_dirs: Option<Vec<(String, String)>>,
 }
 
-pub(crate) fn parse_config_from_file(
-    config_file_path: std::path::PathBuf,
-) -> Result<NodeConfig, FaaSError> {
-    let file_content = std::fs::read(config_file_path)?;
-    let config: RawCoreModulesConfig =
-        from_slice(&file_content).map_err(|err| FaaSError::ConfigParseError(format!("{}", err)))?;
-
+/// Prepare config after parsing it from TOML
+pub(crate) fn from_raw_config(config: RawCoreModulesConfig) -> Result<CoreModulesConfig> {
     let modules_config = config
         .core_module
         .into_iter()
         .map(|module| {
-            let imports: Option<Vec<(String, String)>> = module.imports.map(|import| {
-                import
-                    .into_iter()
-                    .map(|(import_func_name, host_cmd)| {
-                        (import_func_name, host_cmd.try_into::<String>().unwrap())
-                    })
-                    .collect::<Vec<_>>()
-            });
+            let imports = module
+                .imports
+                .map(|import| {
+                    Ok(import
+                        .into_iter()
+                        .map(|(import_func_name, host_cmd)| {
+                            let host_cmd = host_cmd.try_into::<String>()?;
+                            Ok((import_func_name, host_cmd))
+                        })
+                        .collect::<Result<Vec<_>>>()?)
+                })
+                .map_or(Ok(None), |r: Result<_>| r.map(Some))?;
```

Comment:
My motivation to unwrapping in this iteration was that it is an the early stage of Faas creation, and if config is mailformed, smth went very wrong and Faas can't be created. As I got this new approach correct, you are mapping `Err` to `None` here. So if `host_cmd` can't be parsed we just skip imports section at all.

I think that is could be a source of confusion for users, because with None in imports they would highly likely get errors about unsatisfiable imports of a loaded Wasm module. So I suggest to return Err here instead.

Reply:
> you are mapping Err to None
No, it's just a transposition from `Option<Result<T, E>>` to `Result<Option<T>, E>`. Error is kept. I'll rewrite it using `Option::transpose`, intention will be more clear

Reply:
Sorry, I misreaded, yes, I think it is OK.